File: server/Models/House.cs
using System.ComponentModel.DataAnnotations;

namespace gregslist_dotnet_fullstack.Models;

public class House
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  [Range(0, 100000)] public int? Sqft { get; set; }
  [Range(0, 25)] public int? Bedrooms { get; set; }
  [Range(0, 25)] public int? Bathrooms { get; set; }
  [Url] public string ImgUrl { get; set; }
  [MinLength(0), MaxLength(500)] public string Description { get; set; }
  [Range(0, 10000000)] public int? Price { get; set; }
  [Range(1000, 2025)] public int? Year { get; set; }
  [Range(1, 10)] public int? Levels { get; set; }
  public string CreatorId { get; set; }
  public Account Creator { get; set; }
}

File: client/src/models/House.js
export class House {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.sqft = data.sqft
    this.bedrooms = data.bedrooms
    this.bathrooms = data.bathrooms
    this.imgUrl = data.imgUrl
    this.description = data.description
    this.price = data.price
    this.year = data.year
    this.levels = data.levels
  }
}